usaspending_api/download/filestreaming/download_generation.py

```diff
@@ -128,39 +202,68 @@ def generate_download(download_job: DownloadJob, origination: Optional[str] = No
 
     # push file to S3 bucket, if not local
     if not settings.IS_LOCAL:
-        with tracer.trace(
+        with SubprocessTrace(
             name=f"job.{JOB_TYPE}.download.s3",
+            kind=SpanKind.INTERNAL,
+            service="bulk-download",
+        ) as span:
+            span.set_attributes(
+                {
+                    "service": "bulk-download",
+                    "span_type": "Internal",
+                    "resource": f"s3://{settings.BULK_DOWNLOAD_S3_BUCKET_NAME}",
+                    "message": "Push file to S3 bucket, if not local",
+                }
+            )
+
+        with SubprocessTrace(
+            name=f"job.{JOB_TYPE}.s3.command",
+            kind=SpanKind.SERVER,
             service="bulk-download",
-            resource=f"s3://{settings.BULK_DOWNLOAD_S3_BUCKET_NAME}",
-            span_type=SpanTypes.WORKER,
-        ) as span, tracer.trace(
-            name="s3.command",
-            service="aws.s3",
-            resource=".".join(
-                [multipart_upload.__module__, (multipart_upload.__qualname__ or multipart_upload.__name__)]
-            ),
-            span_type=SpanTypes.WEB,
         ) as s3_span:
+            s3_span.set_attributes(
+                {
+                    "service": "aws.s3",
+                    "span_type": "WEB",
+                    "resource": ".".join(
+                        [multipart_upload.__module__, (multipart_upload.__qualname__ or multipart_upload.__name__)]
+                    ),
+                }
+            )
+
             # NOTE: Traces still not auto-picking-up aws.s3 service upload activity
             # Could be that the patches for boto and botocore don't cover the newer boto3 S3Transfer upload approach
-            span.set_tag("file_name", file_name)
+            # span.set_attribute("file_name", file_name)
```

Comment:
Do we capture the `file_name` elsewhere? If so can we remove this line, and maybe the comment above as well?

Reply:
Took out the filename comment but left the earlier comment.

 I did not touch any existing functionality with the bulk-downloader, I added tracing, therefore I do not feel comfortable taking that comment out since it refers to something implementation related on how the download functions.